### PR TITLE
Refactor for a clearer directory structure

### PR DESCRIPTION
**Pull request by vincekurtz**
This follows @amcastro-tri's suggested layout for the most part, with some naming changes make more sense to me. 

Now the structure is
- examples
    - acrobot
    - airhockey
    - ...
- optimizer
    - test
- scripts
- tools
- utils

For example, the main solver is in `optimizer` instead of `traj_opt` since that's a bit more specific (this whole repo is about traj_opt). I also got rid of some of the tiny old examples that we just used for development/debugging.

- - -
This change is [Reviewable](https://reviewable.io/reviews/ToyotaResearchInstitute/idto/2)

### CODE REVIEW

File: examples/allegro_hand/allegro_hand.cc [hunk 1]
```diff
@@ -132,7 +131,7 @@ class AllegroHandExample : public TrajOptExample {
 
     // Add a model of the hand
     std::string sdf_file = idto::FindIDTOResourceOrThrow(
-        "idto/traj_opt/examples/models/allegro_hand.sdf");
+        "examples/models/allegro_hand.sdf");
```
Comment (amcastro-tri) [inline examples/allegro_hand/allegro_hand.cc:134]: in Drake all file paths must start with `drake/`. 
I wonder if we should adopt that style and make them start with `idto`?
That'd also help disambiguate at call site if the model comes from Drake or from Idto?

File: examples/allegro_hand/allegro_hand.cc [hunk 2]
```diff
@@ -132,7 +131,7 @@ class AllegroHandExample : public TrajOptExample {
 
     // Add a model of the hand
     std::string sdf_file = idto::FindIDTOResourceOrThrow(
```
Comment (amcastro-tri) [inline examples/allegro_hand/allegro_hand.cc:133]: nit, we should probably rename to `FindIdtoResourceOrThrow()` to match the Drake style guide (acronysms are also camel cased)

**Review by amcastro-tri — CHANGES_REQUESTED** [review]

File: examples/allegro_hand/allegro_hand.cc (hunk 2 above)
Comment (vincekurtz) [reply examples/allegro_hand/allegro_hand.cc:133]: sgtm, will do

File: examples/allegro_hand/allegro_hand.cc (hunk 1 above)
Comment (vincekurtz) [reply examples/allegro_hand/allegro_hand.cc:134]: Yeah, that would look cleaner. Not totally sure how to make that happen though, I can look into it

File: examples/airhockey/airhockey.cc [hunk 3]
```diff
@@ -1,15 +1,13 @@
...
-#include "drake/multibody/parsing/parser.h"
-#include "drake/multibody/plant/multibody_plant.h"
-#include "drake/multibody/tree/planar_joint.h"
-#include "drake/multibody/tree/prismatic_joint.h"
-#include "drake/multibody/tree/revolute_joint.h"
-#include "idto/common/find_resource.h"
-#include "idto/common/profiler.h"
+#include <drake/multibody/parsing/parser.h>
+#include <drake/multibody/plant/multibody_plant.h>
+#include <drake/multibody/tree/planar_joint.h>
```
[the remark this answers is not in the document]
Comment (vincekurtz) [reply examples/airhockey/airhockey.cc:5]: not a requirement, but I thought this was the c++ convention?

File: BUILD.bazel [hunk 4]
```diff
@@ -9,12 +9,12 @@ load("//tools:shared_library.bzl", "idto_cc_shared_library")
 
 idto_cc_shared_library(
     name = "idto_shared_library",
-    deps = ["//idto/common", "//idto/traj_opt"],
+    deps = ["//utils", "//optimizer"],
```
[the remark this answers is not in the document]
Comment (vincekurtz) [reply BUILD.bazel:12]: Just personal preference :) And made it easier to keep track of the restructuring. We can go back to `common` if you prefer

**Review by amcastro-tri — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/ToyotaResearchInstitute/idto/2)* status: 0 of 131 files reviewed, 7 unresolved discussions (waiting on @vincekurtz)

*[a discussion](https://reviewable.io/reviews/ToyotaResearchInstitute/idto/2#-NfSm2cY9y8HUMYHI9Jv:-NfSm2cY9y8HUMYHI9Jw:b9oslgc) (no related file):*
Reviewable comment test

___

**Review by amcastro-tri — CHANGES_REQUESTED** [review]
I like this! :lgtm: 

> Reviewed 115 of 131 files at r1, 16 of 16 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/ToyotaResearchInstitute/idto/2#-:-NflS3S-Cr7JS5tKujFF:bpzt8h6)* status: all files reviewed, 4 unresolved discussions (waiting on @vincekurtz)

___
*[`-- commits` line 3 at r2](https://reviewable.io/reviews/ToyotaResearchInstitute/idto/2#-NflR2rVAeLtPqY6AG72):*
let's squash and merge into a single commit with clear message for a nice commit history
___
*[`BUILD.bazel` line 12 at r1](https://reviewable.io/reviews/ToyotaResearchInstitute/idto/2#gh-1340556828:-NflPytBEo5WgGQy-vB7:b4fn70l) ([raw file](https://github.com/ToyotaResearchInstitute/idto/blob/78304d02e1be74c0103686372845c1bfcc18780f/BUILD.bazel#L12)):*
<details><summary><i>Previously, vincekurtz (Vince Kurtz) wrote…</i></summary><blockquote>

Just personal preference :) And made it easier to keep track of the restructuring. We can go back to `common` if you prefer
</blockquote></details>

Asking cause Andrew thought `common` would collide wiht Drake's `common`.
I think `utils` is ok.
___
*[`examples/BUILD.bazel` line 7 at r1](https://reviewable.io/reviews/ToyotaResearchInstitute/idto/2#gh-1340555832:-NflP7j7-GUkEZ-zDXHu:b-rqj0a4) ([raw file](https://github.com/ToyotaResearchInstitute/idto/blob/78304d02e1be74c0103686372845c1bfcc18780f/examples/BUILD.bazel#L7)):*
<details><summary><i>Previously, vincekurtz (Vince Kurtz) wrote…</i></summary><blockquote>

Not really, they're just little Drake systems that we use to run MPC, so I feel like they fit with the examples. If you have strong feelings the other way it'd be easy to change
</blockquote></details>

no strong, just a asking. This is good.
___
*[`examples/airhockey/airhockey.cc` line 5 at r1](https://reviewable.io/reviews/ToyotaResearchInstitute/idto/2#gh-1340558534:-NflPH4G01RH5MQ0lfLz:b-70j555) ([raw file](https://github.com/ToyotaResearchInstitute/idto/blob/78304d02e1be74c0103686372845c1bfcc18780f/examples/airhockey/airhockey.cc#L5)):*
<details><summary><i>Previously, vincekurtz (Vince Kurtz) wrote…</i></summary><blockquote>

not a requirement, but I thought this was the c++ convention?
</blockquote></details>

Is it?
I like the IDTO headers only use quotes, so they are easy to spot.
___
*[`optimizer/test/trajectory_optimizer_test.cc` line 25 at r2](https://reviewable.io/reviews/ToyotaResearchInstitute/idto/2#-NflRvl19-XJic7JLssB:-NflRvl19-XJic7JLssC:bbs5xgx) ([raw file](https://github.com/ToyotaResearchInstitute/idto/blob/d9d2f6c0f2281b2df075586873d9a7ebce98ee43/optimizer/test/trajectory_optimizer_test.cc#L25)):*
> ```cpp
> 
> #define PRINT_VAR(a) std::cout << #a ": " << a << std::endl;
> #define PRINT_VARn(a) std::cout << #a ":\n" << a << std::endl;
> ```

are these PRINT_VAR used? consider removingnn
___
*[`utils/find_resource.cc` line 25 at r2](https://reviewable.io/reviews/ToyotaResearchInstitute/idto/2#-NflPqh80846Eu2PsMXk:-NflPqh80846Eu2PsMXl:b-10gtzk) ([raw file](https://github.com/ToyotaResearchInstitute/idto/blob/d9d2f6c0f2281b2df075586873d9a7ebce98ee43/utils/find_resource.cc#L25)):*
> ```cpp
>   if (idto_resource_path.substr(0, 5) != "idto/") {
>     throw std::runtime_error(fmt::format(
>         "Resource path '{}' must start with 'idto/' ", idto_resource_path));
> ```

btw, nice!

**Review by amcastro-tri — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/ToyotaResearchInstitute/idto/2)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @vincekurtz)

**Review by amcastro-tri — APPROVED** [review]

**Review by amcastro-tri — APPROVED** [review]

**Pull request merged**